django_mongodb/indexes.py

```diff
@@ -0,0 +1,100 @@
+from django.core.exceptions import EmptyResultSet, FullResultSet
+from django.db import NotSupportedError
+from django.db.models import Index
+from django.db.models.expressions import Col
+from django.db.models.fields.related_lookups import In
+from django.db.models.lookups import BuiltinLookup
+from django.db.models.sql.query import Query
+from django.db.models.sql.where import AND, XOR, WhereNode
+
+from .query_utils import process_rhs
+
+mongo_operators_idx = {
+    "exact": lambda a, b: {a: {"$eq": b}},
+    "gt": lambda a, b: {a: {"$gt": b}},
+    "gte": lambda a, b: {a: {"$gte": b}},
+    "lt": lambda a, b: {a: {"$lt": b}},
+    "lte": lambda a, b: {a: {"$lte": b}},
+    "in": lambda a, b: {a: {"$in": b}},
+}
+
+
+def _get_condition_mql(self, model, schema_editor):
+    """Analogous to Index._get_condition_sql()."""
+    query = Query(model=model, alias_cols=False)
+    where = query.build_where(self.condition)
+    compiler = query.get_compiler(connection=schema_editor.connection)
+    return where.as_mql_idx(compiler, schema_editor.connection)
+
+
+def builtin_lookup_idx(self, compiler, connection):
+    if not isinstance(self.lhs, Col):
+        raise NotSupportedError("Expressions as indexes are not supported.")
+    lhs_mql = self.lhs.target.column
+    value = process_rhs(self, compiler, connection)
+    try:
+        return mongo_operators_idx[self.lookup_name](lhs_mql, value)
```

Comment:
This implementation has a problem with tests constraints in Django, I think we have two options:

1. Modify the unit test  in order not to use `isnull` index. 
2. Change the isnull behaviour.

I think the option 1 is better.

Reply:
I amended `test_is_null_condition` on the Django fork to reflect the fact that it raises `NotSupportedError`.